### PR TITLE
Resample refactor

### PR DESCRIPTION
**Pull request by jdavies-st**
Refactor of resample that does the following:

- move drizpars reference file parsing to the top level Step function
- remove the NIRCam-, NIRISS- and MIRI-specific drizpars datamodels and schemas.  They aren't used or needed.  The standard drizpars datamodel works for all instruments.
- remove the NIRCam- and MIRI-specific outlierpars datamodels and schemas.  They aren't used or needed.  The standard outlierpars datamodel works for all instruments.
- change `wht_type` to `weight_type` in resample
- subclass ResampleSpecStep from ResampleStep so they can share attributes and methods that are common.
- fix bugs in parameter passing between top-level ResampleStep and the underlying classes and functions that do that actual work.

### CODE REVIEW

**Comment by jdavies-st** [comment]
Resolves #2243.

**Comment by jdavies-st** [comment]
Resolves #2218.

**Comment by jdavies-st** [comment]
Addresses the problems with `resample` in issue #2059, but that issue should remain open as the `pathloss` step now fails for that dataset.

File: jwst/resample/resample_spec_step.py [hunk 1]
```diff
@@ -44,6 +33,13 @@ def process(self, input):
         for reftype in self.reference_file_types:
             ref_filename = self.get_reference_file(input_models[0], reftype)
 
+        if ref_filename != 'N/A':
+            self.log.info('Drizpars reference file: {}'.format(ref_filename))
+            kwargs = self.get_drizpars(ref_filename, input_models)
+        else:
+            # Deal with NIRSpec which currently has no default drizpars reffile
```
Comment (hbushouse) [inline jwst/resample/resample_spec_step.py]: I assume this is a temporary workaround until someone creates a drizpars reffile for NIRSpec?

File: jwst/resample/resample_utils.py [hunk 2]
```diff
@@ -171,21 +171,19 @@ def _reproject(x, y):
...
-    if wht_type.lower()[:3] == 'err':
+    if weight_type == 'err':
         err_model = np.nan_to_num(model.err)
         inwht = (exptime / err_model)**2 * dqmask
         log.debug("DEBUG weight mask: {} {}".format(type(inwht), np.sum(inwht)))
-    # elif wht_type == 'IVM':
+    # elif weight_type == 'ivm':
     #     _inwht = img.buildIVMmask(chip._chip,dqarr,pix_ratio)
-    elif wht_type.lower()[:3] == 'exp':
+    elif weight_type == 'exp':
```
Comment (hbushouse) [inline jwst/resample/resample_utils.py]: Isn't the value that gets passed in for exposure time weighting "exptime", as opposed to "exp"? Or is that reset somewhere in the param parsing?

Comment (jdavies-st) [reply jwst/resample/resample_utils.py]: Ah, thanks for catching that!  That's a bug.

File: jwst/resample/resample_spec_step.py (hunk 1 above)
Comment (jdavies-st) [reply jwst/resample/resample_spec_step.py]: Yes, it also is a general "if there is no reffile" workaround for any undefined modes that don't have a reffile.  NIRSpec is the one that I ran into that did not have a reffile, but perhaps there are others.  NIRCam, MIRI and NIRISS have reffiles right now.  So perhaps FGS needs one too?  Not sure they'll be resampling FGS science data.

File: jwst/resample/resample_utils.py (hunk 2 above)
Comment (hbushouse) [reply jwst/resample/resample_utils.py]: minion_what

**Pull request merged**